File: CSharpFundamentals/CSharpOOPbasics/InheritanceEx/Mankind/Student.cs
using System;
using System.Text;
using System.Text.RegularExpressions;

namespace Mankind
{
    public class Student : Human
    {
        private const int FACULTY_MIN = 5;
        private const int FACULTY_MAX = 10;
        private string facultyNumber;

        public Student(string firstName, string lastName, string facultyNumber)
    : base(firstName, lastName)
        {
            this.FacultyNumber = facultyNumber;
        }

        public string FacultyNumber
        {
            get { return facultyNumber; }
            private set
            {
                string pattern = @"^[A-Za-z0-9]{5,10}$";
                if (!Regex.IsMatch(value, pattern))
                {
                    throw new ArgumentException("Invalid faculty number!");
                }
                facultyNumber = value;
            }
        }

        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();
            sb.AppendLine(base.ToString());
            sb.AppendLine($"Faculty number: {this.FacultyNumber}");
            return sb.ToString().TrimEnd();
        }
    }
}

File: JavaDB/DatabaseAdvancedFrameworks/JavaOOPPrinciplesEx/Mankind/models/Student.java
package Mankind.models;

public class Student extends Human {
    private static final int MIN_FAC_NUM = 5;
    private static final int MAX_FAC_NUM = 10;

    private String facultyNumber;

    public Student(String firstName, String lastName, String facultyNumber ) {
        super(firstName, lastName);
        this.setFacultyNumber(facultyNumber);
    }

    public String getFacultyNumber() {
        return facultyNumber;
    }

    public void setFacultyNumber(String facultyNumber) {
        if (facultyNumber.length() < MIN_FAC_NUM || facultyNumber.length() > MAX_FAC_NUM){
            throw new IllegalArgumentException(
                    "Invalid faculty number!"
            );
        }
        this.facultyNumber = facultyNumber;
    }

    @Override
    public String toString(){
        return super.toString() +
                String.format("Faculty number: %s%n", this.getFacultyNumber());
    }
}
